### ISSUE TITLE
serialization of buffers sizer/pack generate different sizes for null buffers

### ISSUE THREAD

**Issue by jjwilke** [sstsimulator/sst-core#240]
````  
    template <typename Int>
    void
    binary(void*& buffer, Int& size){
       .....      
       case SIZER: {
            sizer_.add(sizeof(Int));
            sizer_.add(size);
            break;
        }
````
given a null buffer with size 0, this gives size=sizeof(int)=4

````  
    template <typename Int>
    void
    binary(void*& buffer, Int& size){
       .....      
        case PACK: {
            packer_.pack(size);
            packer_.pack_buffer(buffer, size);
            break;
        }

void
ser_packer::pack_buffer(void* buf, int size)
{
  if (buf && size != 0){
    char* charstr = next_str(size);
    ::memcpy(charstr, buf, size);
  } else {
    int nullsize = 0;
    pack(nullsize);
  }
}
````

Packer packs size=4 in the case statement then packs nullsize in pack_buffer for a total size of 8.

Program then crashes with ser_buffer_overrun

**Comment by jjwilke** [comment]
This might have been my bug in the first place from first sprockit version.

**Comment by nmhamster** [comment]
@jjwilke - can you confirm whether this really is a serializer bug or whether its in Sprockit or both? Thanks.

**Comment by nmhamster** [comment]
@jjwilke - any update on this issue?

**Comment by jjwilke** [comment]
My fix was accepted as pull request #241
